react-components/src/utils/hooks.js

```diff
@@ -94,21 +94,61 @@ export const useStorefrontEvents = () => {
 };
 
 export const usePageType = () => {
-    // Currently only using CMS, Category and Product. Added other types for completeness
-    // and to be used in the future.
     const PageTypes = {
+        // Pages with landing-page template or pages that match the store root URL
+        // as defined in the store config.
         CMS: 'CMS',
+
+        // Pages with category component or category-page template
         CATEGORY: 'Category',
+
+        // Pages with product component or product-page template
         PRODUCT: 'Product',
+
+        // Pages with cart component
         CART: 'Cart',
-        CHECKOUT: 'Checkout'
+
+        // Pages with checkout component
+        CHECKOUT: 'Checkout',
+
+        // Any other pages
+        PAGE_BUILDER: 'PageBuilder'
     };
 
-    if (document.querySelector('[data-cif-product-context]')) {
+    // Detect homepage, either by template name or by URL
+    let template = document.querySelector('meta[name="template"]');
+    if (template) {
+        template = template.getAttribute('content');
+    }
+
+    let canonicalUrl = document.querySelector('link[rel="canonical"]');
+    if (canonicalUrl) {
+        canonicalUrl = canonicalUrl.getAttribute('href');
+    }
+
+    let storeRootUrl = null;
+    try {
+        let storeConfig = JSON.parse(document.querySelector('meta[name="store-config"]').getAttribute('content'));
+        storeRootUrl = storeConfig.storeRootUrl;
+    } catch (err) {
+        // Could not parse store config, ignore for now
+    }
+
+    if (template == 'landing-page' || (canonicalUrl && storeRootUrl && canonicalUrl == storeRootUrl)) {
```

Comment:
This assumption `canonicalUrl == storeRootUrl` is not correct. 

The storeRootUrl is [mapped per default](https://github.com/adobe/aem-core-cif-components/blob/master/bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/storeconfigexporter/StoreConfigExporterImpl.java#L118) using `ResourceResolve#map(HttpServletRequest, String)` and canonical url using `ResourceResolver#map(String)` (without the request context). 

The storeRootUrl may be a path `/us/en.html` but the canonical url will always be absolute `http://wknd.site/en.html`.

Reply:
This seems to be inconsistent. If you check https://publish-p10603-e75517-cmstg.adobeaemcloud.com/us/en.html, the canonical URL is relativ. When going to any other page, it becomes absolute.

Any idea how else we can check which page is the homepage? Template name alone might not be sufficient, since this is custom and the landing-page template is only a recommendation.

Reply:
This inconsistency is addressed in https://github.com/adobe/aem-cif-guides-venia/pull/245

I guess you could just remove the authority of the canonical url and only compare the paths.